pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2452,12 +2491,20 @@ func (c *Cloud) EnsureLoadBalancer(clusterName string, apiService *api.Service,
 		return nil, err
 	}
 
+	// Add loadbalancer's security group's rules into instances' security groups' rules.
 	err = c.updateInstanceSecurityGroupsForLoadBalancer(loadBalancer, instances)
 	if err != nil {
 		glog.Warningf("Error opening ingress rules for the load balancer to the instances: %v", err)
 		return nil, err
 	}
 
+	// Add sharedSecurityGroupID into instances' rules only.
+	err = c.updateInstanceSharedSecurityGroups(sharedSecurityGroupID, instances)
```

Comment:
Let's put the switching logic that is currently in updateInstanceSecurityGroupsForLoadBalancer and updateInstanceSharedSecurityGroups here (and in any other callers?).